server/src/utils/checkForObjectSelection.ts

```diff
@@ -0,0 +1,13 @@
+import { GraphQLResolveInfo } from 'graphql';
+
+export const checkForObjectSelection = (info: GraphQLResolveInfo, fields: string[]) => {
+  let parent = info.operation.selectionSet;
+  for (let index = 0; index < fields.length; index++) {
+    const field = fields[index];
+    const selectionNode = parent.selections.find(selection=>(selection as any).name.value === field) ?? false;
```

Comment:
This will not work with using `Fragments` as the `name.value` will resolve as the `Fragment` name, i.e `PostSnippet`